### PR TITLE
fix: old aot_tools binaries do not work

### LINKED ISSUE #1644: fix: dart compiler ran out of memory trying to compile linker?
This actually looks like difficulty compiling the aot-tools.dill, not the linker itself running out of memory:

```
✗ Failed to link AOT files: Exception: Failed to link: Exhausted heap space, trying to allocate 35232 bytes.
Crash when compiling $HOME/shorebird/bin/cache/artifacts/aot-tools/9472e50354253fde3f2cdb25d8115f7aa6d3c68a/aot-tools:
Out of Memory

#0      _StringBase._concatRangeNative (dart:core-patch/string_patch.dart:962:78)
#1      _JsonStringStringifier.stringify (dart:convert/json.dart:968:19)
#2      SourceLibraryBuilder.addProblem (package:front_end/src/fasta/source/source_library_builder.dart:1868:52)
#3      SourceLoader.tokenize (package:front_end/src/fasta/source/source_loader.dart:949:24)
#4      _rootRunUnary (dart:async/zone.dart:1407:47)
#5      _CustomZone.runUnary (dart:async/zone.dart:1308:19)
#6      _FutureListener.handleValue (dart:async/future_impl.dart:162:18)
#7      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:846:45)
#8      Future._propagateToListeners (dart:async/future_impl.dart:875:13)
#9      Future._completeWithValue (dart:async/future_impl.dart:647:5)
#10     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:721:7)
#11     _rootRun (dart:async/zone.dart:1399:13)
#12     _CustomZone.run (dart:async/zone.dart:1301:19)
#13     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:1233:23)
#14     _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
#15     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)
#16     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#17     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:405:11)
#18     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:429:5)
#19     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)

 (2250.4s)
 ```

### PR DESCRIPTION
**Pull request by eseidel**
We try to execute them as dart scripts rather than running them
directly.  While I was in here I cleaned up a typo and removed
fileName in favor of name.

Fixes https://github.com/shorebirdtech/shorebird/issues/1644.

### CODE REVIEW

**Review by eseidel — COMMENTED** [review]
I think I still need to add another test.

File: packages/shorebird_cli/lib/src/cache.dart [hunk 1]
```diff
@@ -29,7 +29,7 @@ class CacheUpdateFailure implements Exception {
   String toString() => 'CacheUpdateFailure: $message';
 }
 
-typedef ArchiveExtracter = Future<void> Function(
+typedef ArchiveExtractor = Future<void> Function(
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/cache.dart:32]: 💯

File: packages/shorebird_cli/lib/src/cache.dart [hunk 2]
```diff
@@ -133,19 +133,21 @@ abstract class CachedArtifact {
   final Cache cache;
   final Platform platform;
 
+  /// The on-disk name of the artifact.
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/cache.dart:137]: Thanks for adding the inline comments!

Comment (eseidel) [reply packages/shorebird_cli/lib/src/cache.dart:137]: Honestly it's selfish. :) My future me wants them.

File: packages/shorebird_cli/lib/src/cache.dart [hunk 3]
```diff
@@ -218,7 +217,7 @@ class AotToolsArtifact extends CachedArtifact {
 
   @override
   String get storageUrl =>
-      '${cache.storageBaseUrl}/${cache.storageBucket}/shorebird/${shorebirdEnv.shorebirdEngineRevision}/aot-tools.dill';
+      '${cache.storageBaseUrl}/${cache.storageBucket}/shorebird/${shorebirdEnv.shorebirdEngineRevision}/$name';
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/cache.dart:221]: I think this will still fail when we are using flutter revisions that pre-date the `.dill`.

Comment (eseidel) [reply packages/shorebird_cli/lib/src/cache.dart:221]: Fixing.

**Review by felangel — APPROVED** [review]
LGTM ✅

**Pull request merged**